src/radiation/radiation.cpp

```diff
@@ -0,0 +1,621 @@
+//
+// Created by owen on 3/19/22.
+//
+#include "radiation.hpp"
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <utility>
+#include "environment/runEnvironment.hpp"
+#include "finiteVolume/compressibleFlowFields.hpp"
+#include "utilities/mathUtilities.hpp"
+
+ablate::radiation::RadiationSolver::RadiationSolver(std::string solverId, std::shared_ptr<domain::Region> region, int rayNumber,
+                                                    std::shared_ptr<parameters::Parameters> options)
+    : CellSolver(std::move(solverId), std::move(region),  std::move(options)) {
+    raynumber = rayNumber;
+    nTheta = raynumber; //The DEFAULT number of angles to solve with, should be given by user input probably?
+    nPhi = 2 * raynumber; //The DEFAULT number of angles to solve with, should be given by user input
+}
+
+ablate::radiation::RadiationSolver::~RadiationSolver() {
+}
+
+void ablate::radiation::RadiationSolver::Setup() {  // allows initialization after the subdomain and dm is established
+    ablate::solver::CellSolver::Setup();
+    dim = subDomain->GetDimensions();  // Number of dimensions already defined in the setup
+
+}
+void ablate::radiation::RadiationSolver::Initialize() {
+    // this->reallySolveParallelPlates();
+    this->RayInit();  // Runs the ray initialization finding cell indices
+    /*this->RayTrace(0);
+    for (int i = 1; i < 20; i++) {
+        this->RayProduct(0, i);
+    }*/
+
+}
+
+/// Declaring function for the initialization to call, draws each ray vector and gets all of the cells associated with it (sorted by distance and starting at the boundary working in)
+void ablate::radiation::RadiationSolver::RayInit() {
+    //ablate::solver::CellSolver::Setup();
+
+    double theta;  // represents the actual current angle (inclination)
+    double phi;    // represents the actual current angle (rotation)
+
+    /// Locally get a range of cells that are included in this subdomain at this time step for the ray initialization
+    // March over each cell in this region to create the stencil
+    // IS cellIS;
+    // PetscInt cStart, cEnd;  // Keep these as local variables, the cell indices in the radiation initializer should be updated with every radiation initialization. The variables here are owned by the
+                            // DM
+    stencilSet.clear(); //Clear the existing set because a new set of cells will be created below
+
+    solver::Range cellRange;
+    GetCellRange(cellRange);
+    // std::set<PetscInt> stencilSet; //Moved to a class variable
+    for (PetscInt c = cellRange.start; c < cellRange.end; ++c) {
```

Comment:
Do we need the stencilSet or could we march directly over the cellRange instead?